module/userExtension/helper.js

```diff
@@ -0,0 +1,242 @@
+    static getProfile(queryObject, token) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let profileData = await sunBirdService.getUserProfileInfo(token, queryObject.userId);
+                let roles;
+                profileData = JSON.parse(profileData);
+                if (profileData.responseCode == messageConstants.common.RESPONSE_OK) {
+                    if (profileData.result &&
+                        profileData.result.response
+                    ) {
+
+                        if (profileData.result.response.roles) {
+                            roles = profileData.result.response.roles;
+                        }
+                        await Promise.all(profileData.result.response.organisations.map(orgInfo => {
+                            if (orgInfo.roles) {
+                                roles.push(...orgInfo.roles);
+                            }
+                        }))
+                    }
+                }
+
+                if (roles) {
+                    roles = Array.from(new Set(roles));
+                }
+
+                let userExtDocument = await database.models.userExtension.findOne(
+                    queryObject,
+                    {
+                        updatedBy: 0,
+                        createdBy: 0,
+                        createdAt: 0,
+                        updatedAt: 0,
+                        status: 0,
+                        deleted: 0,
+                        userId: 0,
+                        "__v": 0
+                    }
+                ).lean();
+
+                if (userExtDocument) {
+                    userExtDocument.roles = userExtDocument.roles.map(role => role.code)
+                    if (roles) {
+                        userExtDocument.roles.push(...roles);
+                    }
+                }
+                return resolve(userExtDocument);
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+
+    }
+
+
+    /**
+ * create.
+ * @method
+ * @name  create
+ * @param  {request, token, userId}  - requested body, token and userId.
+ * @returns {json} Response consists of created user details.
+ */
+    static create(request, token, userId) {
+
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (request.organisation &&
+                    request.organisation.value &&
+                    request.userName && request.password) {
+                    let response = await sunBirdService.createUser(request, token);
+                    if (response && response.responseCode == messageConstants.common.RESPONSE_OK) {
+
+
+                        let rolesId = [];
+                        let organisationsRoles = [];
+                        let plaformRoles = [];
+                        // let allRoles = [];
+
+
+                        let rolesDocuments = await database.models.platformRolesExt.find({}, {
```

Comment:
@rakeshSgr , @deepakr123 - Lets not call other database models directly, always use their module helpers.